### ISSUE TITLE
Time & duration input fields are not labeled

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#14144]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

For time and duration inputs that use the `ha-base-time-input` element, the individual fields are not labeled, and the fields are not grouped together programmatically and associated with the actual label.  This makes them difficult to use with assistive technology like screen readers and has other accessibility concerns for cognitive disabilities.

### Describe the behavior you expected

Individual fields should be labeled (e.g. by a tooltip), and the fields should be grouped and labeled using either an [HTML `fieldset`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) or the [ARIA `group` role](https://www.w3.org/TR/WCAG20-TECHS/ARIA17.html).

### Steps to reproduce the issue

n/a

### What version of Home Assistant Core has the issue?

2022.10.x

### In which browser are you experiencing the issue with?

n/a

### Which operating system are you using to run this browser?

n/a

**Comment by steverep** [comment]
Still an issue

**Comment by gayanMatch** [comment]
Hi @steverep is this still an issue? if so, I'd like to work on it

**Comment by allenlarkins** [comment]
Hi @steverep is this still an issue? if so, I'd like to work on it
